cpp/src/parquet/encoding.cc

```diff
@@ -1484,6 +1521,41 @@ class DictDecoderImpl : public DecoderImpl, virtual public DictDecoder<Type> {
   // Perform type-specific initiatialization
   void SetDict(TypedDecoder<Type>* dictionary) override;
 
+  template <typename T = Type,
+            typename = std::enable_if_t<std::is_same_v<T, ByteArrayType> ||
+                                        std::is_same_v<T, LargeByteArrayType>>>
+  void SetByteArrayDict(TypedDecoder<Type>* dictionary) {
+    DecodeDict(dictionary);
+
+    auto dict_values = reinterpret_cast<ByteArray*>(dictionary_->mutable_data());
+
+    using offset_type = typename EncodingTraits<Type>::ArrowType::offset_type;
+
+    int32_t total_size = 0;
```

Comment:
By the way, can we use `offset_type` to store `total_size`?

Reply:
I guess we can. In this case, we might as well just remove the `AddWithOverflow` call right